packages/rocketchat-oembed/server/server.coffee

```diff
@@ -13,73 +8,50 @@ getUrlContent = (urlObj, redirectCount = 5, callback) ->
 	if _.isString(urlObj)
 		urlObj = URL.parse urlObj
 
-	opts =
-		method: 'GET'
-		port: urlObj.port
-		hostname: urlObj.hostname
-		path: urlObj.path
-		rejectUnauthorized: !RocketChat.settings.get 'Allow_Invalid_SelfSigned_Certs'
-		headers:
-			'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2227.0 Safari/537.36'
-
-	httpOrHttps = if urlObj.protocol is 'https:' then https else http
-
 	parsedUrl = _.pick urlObj, ['host', 'hash', 'pathname', 'protocol', 'port', 'query', 'search']
 
 	RocketChat.callbacks.run 'oembed:beforeGetUrlContent',
-		requestOptions: opts
+		urlObj: urlObj
 		parsedUrl: parsedUrl
 
-	request = httpOrHttps.request opts, Meteor.bindEnvironment (response) ->
-		if response.statusCode in [301,302,307] and response.headers.location?
-			request.abort()
-			console.log response.headers.location
-
-			if redirectCount <= 0
-				return callback null, {parsedUrl: parsedUrl}
+	url = URL.format urlObj
+	opts =
+		url: url
+		strictSSL: !RocketChat.settings.get 'Allow_Invalid_SelfSigned_Certs'
+		gzip: true
+		maxRedirects: redirectCount
+		headers:
+			'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2227.0 Safari/537.36'
 
-			getUrlContent response.headers.location, --redirectCount, callback
-			return
+	headers = null
+	chunks = []
+	chunksTotalLength = 0
 
+	stream = request opts
+	stream.on 'response', (response) ->
 		if response.statusCode isnt 200
-			return callback null, {parsedUrl: parsedUrl}
-
-		chunks = []
-		chunksTotalLength = 0
-		response.on 'data', (chunk) ->
-			chunks.push chunk
-			chunksTotalLength += chunk.length
-			if chunksTotalLength > 250000
-				request.abort()
-
-		response.on 'end', Meteor.bindEnvironment ->
-			buffer = Buffer.concat(chunks)
-
-			try
-				if response.headers['content-encoding'] is 'gzip'
-					buffer = gunzipSync buffer
-				else if response.headers['content-encoding'] is 'deflate'
-					buffer = inflateSync buffer
-
-			callback null, {
-				headers: response.headers
-				body: buffer.toString()
-				parsedUrl: parsedUrl
-			}
-
-		response.on 'error', (error) ->
-			callback null, {
-				error: error
-				parsedUrl: parsedUrl
-			}
-
-	request.on 'error', (error) ->
+			return stream.abort()
+		headers = response.headers
+
+	stream.on 'data', (chunk) ->
+		chunks.push chunk
+		chunksTotalLength += chunk.length
+		if chunksTotalLength > 250000
+			stream.abort()
+
+	stream.on 'end', Meteor.bindEnvironment ->
+		buffer = Buffer.concat(chunks)
```

Comment:
Yes. The request module pipe gunzip for each chunk. [see also](https://github.com/request/request/blob/master/request.js#L924). I think that this library is made well.
My Test:
[gziped site](http://checkgzipcompression.com/?url=http%3A%2F%2Fcheckgzipcompression.com%2F)
Screenshot:
![screenshot-localhost 3000 2016-03-23 07-04-02](https://cloud.githubusercontent.com/assets/9736483/13969501/a5295894-f0c6-11e5-8aeb-73a6d368ed8b.png)